src/semi_canonicalize.cc

```diff
@@ -75,10 +75,22 @@ void SemiCanonical::startup() {
     Ua_ = SharedMatrix(new Matrix("Ua", nmopi_, nmopi_));
     Ub_ = SharedMatrix(new Matrix("Ub", nmopi_, nmopi_));
 
+    Ua_->identity();
+    Ub_->identity();
+
     // Preapare orbital rotation matrix, which transforms only active MOs
     Ua_t_ = ambit::Tensor::build(ambit::CoreTensor, "Ua", {nact_, nact_});
     Ub_t_ = ambit::Tensor::build(ambit::CoreTensor, "Ub", {nact_, nact_});
 
+    Ua_t_.iterate([&](const std::vector<size_t>& i, double& value) {
+        if (i[0] == i[1])
+            value = 1.0;
+    });
+    Ub_t_.iterate([&](const std::vector<size_t>& i, double& value) {
+        if (i[0] == i[1])
+            value = 1.0;
+    });
+
```

Comment:
Are these loops doing something different than what is accomplished by the function `->identity();`?